src/System.Numerics.Tensors/System/Numerics/TensorArithmetic.cs

```diff
@@ -12072,28 +12716,33 @@ public void Contract(DenseTensor<uint> left, DenseTensor<uint> right, int[] left
                     int leftIndex = leftIndexNonSumming + leftIndexSumming;
                     int rightIndex = rightIndexNonSumming + rightIndexSumming;
 
-                    sum += (uint)(left.Buffer[leftIndex] * right.Buffer[rightIndex]);
+                    sum += (uint)(leftSpan[leftIndex] * rightSpan[rightIndex]);
                 }
 
-                result.Buffer[resultIndex] = sum;
+                resultSpan[resultIndex] = sum;
             }
         }
         public void Decrement(DenseTensor<uint> tensor, DenseTensor<uint> result)
         {
 
+            var resultSpan = result.Buffer.Span;
+            var tensorSpan = tensor.Buffer.Span;
             for(int i = 0; i < result.Length; i++)
```

Comment:
It might be worth changing the loop to use resultSpan.Length. The JIT is doing some bounds checks optimizations that I am not sure would kick in if you are not using the length of the same thing as you loop over.

Reply:
Typically I'll be accessing more than one span inside the loop and all are constrained by the same bounds check.  Any way to make the JIT aware of this?